x-pack/plugins/cases/server/client/attachments/delete.ts

```diff
@@ -79,10 +75,131 @@ export async function deleteAll(
   }
 }
 
+export const deleteFileAttachments = async (
+  { caseId, fileIds }: DeleteFileArgs,
+  clientArgs: CasesClientArgs
+) => {
+  const {
+    user,
+    services: { attachmentService, userActionService },
+    logger,
+    authorization,
+    fileService,
+  } = clientArgs;
+
+  try {
+    const fileEntities = await getFileEntities(caseId, fileIds, fileService);
+
+    // It's possible for this to return an empty array if there was an error creating file attachments in which case the
+    // file would be present but the case attachment would not
+    const fileAttachments = await attachmentService.getter.getFileAttachments({ caseId, fileIds });
+
+    await authorization.ensureAuthorized({
+      entities: [
+        ...fileAttachments.map((attachment) => ({
+          id: attachment.id,
+          owner: attachment.attributes.owner,
+        })),
+        ...fileEntities,
+      ],
+      operation: Operations.deleteComment,
+    });
+
+    await Promise.all([
+      pMap(fileIds, async (fileId: string) => fileService.delete({ id: fileId }), {
+        concurrency: MAX_CONCURRENT_SEARCHES,
+      }),
+      attachmentService.bulkDelete({
+        attachmentIds: fileAttachments.map((so) => so.id),
+        refresh: false,
+      }),
+    ]);
+
+    await userActionService.creator.bulkCreateAttachmentDeletion({
+      caseId,
+      attachments: fileAttachments.map((attachment) => ({
+        id: attachment.id,
+        owner: attachment.attributes.owner,
+        attachment: attachment.attributes,
+      })),
+      user,
+    });
+  } catch (error) {
+    throw createCaseError({
+      message: `Failed to delete file attachments for case: ${caseId} file ids: ${fileIds}: ${error}`,
+      error,
+      logger,
+    });
+  }
+};
+
+const getFileEntities = async (
+  caseId: DeleteFileArgs['caseId'],
+  fileIds: DeleteFileArgs['fileIds'],
+  fileService: FileServiceStart
+) => {
+  const files = await getFiles(caseId, fileIds, fileService);
+
+  const fileEntities = createFileEntities(files);
+
+  return fileEntities;
+};
+
+type FileWithRequiredCaseMetadata = Omit<File<CaseFileMetadata>, 'data'> & {
+  data: Omit<FileJSON<CaseFileMetadata>, 'meta'> & { meta: CaseFileMetadata };
+};
+
+const getFiles = async (
+  caseId: DeleteFileArgs['caseId'],
+  fileIds: DeleteFileArgs['fileIds'],
+  fileService: FileServiceStart
+) => {
+  // it's possible that we're trying to delete a file when an attachment wasn't created (for example if the create
+  // attachment request failed)
+  const files = await pMap(fileIds, async (fileId: string) => fileService.getById({ id: fileId }), {
+    concurrency: MAX_CONCURRENT_SEARCHES,
+  });
+
+  const [validFiles, invalidFiles] = partition(files, (file) => {
+    return CaseFileMetadataRt.is(file.data.meta) && file.data.meta.caseId === caseId;
```

Comment:
@cnasikas should we be more lenient here? For example lets say a user manually created a file and formatted the metadata incorrectly. If we are going to prevent deletion through the file management page, the user wouldn't have anyway to delete the file.

Maybe instead we just check that the metadata contains the `caseId` field and ignores the format of the owner since this function doesn't actually need it.

We can still enforce that the caseId has to be present and match the passed in `caseId` from the request.